platforms/core-configuration/model-core/src/main/java/org/gradle/internal/instantiation/generator/AsmBackedClassGenerator.java

```diff
@@ -1951,7 +1952,7 @@ private static Class<?> rawTypeParam(PropertyMetadata property, int paramNum) {
         if (argument instanceof Class) {
             return (Class<?>) argument;
         }
-        return (Class<?>) ((ParameterizedType) argument).getRawType();
+        return (Class<?>) Cast.cast(ParameterizedType.class, argument).getRawType();
```

Comment:
❔ why do we need this change?

Reply:
Late, but this gives a better error message when argument is not a ParameterizedType